### PR TITLE
fix: Multiple changes to the action based on testing from an example repository

### PR DESCRIPTION
**Pull request by bradegler**
A number of issues were discovered while updating the example in an external repo mainly:

* Authentication data was being overwritten with the checkout step
* Environment variable behavior was not working as expected leading to multiple variables being null that caused steps to fail
* Passing outputs between steps did not work as expected resulting in null data being passed

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -97,25 +89,30 @@ runs:
...
     - name: 'add-comment'
       env:
-        training_jobs: '${{ steps.vertex-execution.outputs.training_jobs }}'
+        vertex_job_uri: 'https://console.cloud.google.com/vertex-ai/locations'
+        vertex_notebook_uri: 'https://notebooks.cloud.google.com/view'
+        region: '${{ inputs.region }}'
       uses: 'actions/github-script@9ac08808f993958e9de277fe43a64532a609130e'
       with:
         script: |
+          const fs = require('fs');
```
[the remark this answers is not in the document]
Comment (verbanicm) [reply action.yml:109]: Can see example here: https://github.com/google-github-actions/.github/blob/main/.github/workflows/draft-release.yml

[the remark this answers is not in the document]
Comment (bradegler) [reply action.yml]: Good idea. I added the try / catch

**Review by verbanicm — APPROVED** [review]

File: action.yml [hunk 2]
```diff
@@ -27,32 +27,21 @@ inputs:
...
     default: 'us-central1'
-
-permissions:
-  pull-requests: 'write'
-  issues: 'write'
-  contents: write
-  id-token: 'write'
-
-env:
-  VERTEX_CONTAINER: 'gcr.io/deeplearning-platform-release/base-cu110:latest'
```
Comment (sethvargo) [inline action.yml:41]: Do users need to know these values now, or is there a sane default?

**Review by sethvargo — APPROVED** [review]

File: action.yml (hunk 2 above)
Comment (bradegler) [reply action.yml:41]: They are set to a sane default. Defining them in an 'env' section didn't seem to work - either there is some quirk with this in composite actions or I'm doing something wrong. I could move them to the inputs section but they would always use the default and may lead to confusion.

Comment (bradegler) [reply action.yml:41]: The notebook executor is in that container. The only reason we'd change it is if Vertex switched to a new container. Same for the URLs, those are Vertex's defined URLs, not something we should expect a user to update.

**Pull request merged**